lib/tapioca/compilers/dsl/url_helpers.rb

```diff
@@ -0,0 +1,80 @@
+# typed: true
+# frozen_string_literal: true
+
+require "parlour"
+
+begin
+  require "rails"
+rescue LoadError
+  return
+end
+
+module Tapioca
+  module Compilers
+    module Dsl
+      class UrlHelpers < Base
+        extend T::Sig
+
+        sig { override.params(root: Parlour::RbiGenerator::Namespace, constant: T.class_of(Module)).void }
+        def decorate(root, constant)
+          path_helper_methods = GeneratedPathHelpersModule.instance_methods(false)
+          url_helper_methods = GeneratedUrlHelpersModule.instance_methods(false)
+
+          if constant == GeneratedPathHelpersModule
+            generate_module_for(root, GeneratedPathHelpersModule, path_helper_methods)
+          elsif constant == GeneratedUrlHelpersModule
+            generate_module_for(root, GeneratedUrlHelpersModule, url_helper_methods)
+          else
+            root.path(constant) do |mod|
+              create_mixins_for(mod, constant, GeneratedUrlHelpersModule)
+              create_mixins_for(mod, constant, GeneratedPathHelpersModule)
+            end
+          end
```

Comment:
This would be much nicer as a `case/when` block:
```suggestion
          case constant
          when GeneratedPathHelpersModule, GeneratedUrlHelpersModule
            generate_module_for(root, constant)
          else
            root.path(constant) do |mod|
              create_mixins_for(mod, constant, GeneratedUrlHelpersModule)
              create_mixins_for(mod, constant, GeneratedPathHelpersModule)
            end
          end
        end
```